preppal-fe/src/pages/profile.tsx

```diff
@@ -26,11 +31,23 @@ function Profile() {
                         "x-auth-token": token
                     }
                 };
-                const res = await fetch("http://localhost:9001/api/auth/", req).then(res => res.json());
-                setName(res.username);
-                setBio(res.bio);
-                setFollowing(res.following);
-                setFollowingCount(userFollowing.length)
+                let res;
+                if (myProfile)
+                {
+                    res = await fetch("http://localhost:9001/api/auth/", req).then(res => res.json());
+                    setName(res.username);
+                    setBio(res.bio);
+                    setFollowing(res.following);
+                    setFollowingCount(userFollowing.length)
+                }
+                else
+                {
```

Comment:
this block should be outside the token since we shouldn't need to be logged in to view author's profile.

it should go if(myProfile) {
if token{}
}
else {
}